bindings/rust/extended/s2n-tls/src/connection.rs

```diff
@@ -1319,6 +1320,46 @@ impl Connection {
         unsafe { s2n_connection_is_session_resumed(self.connection.as_ptr()) == 1 }
     }
 
+    /// Append an external psk to a connection.
+    ///
+    /// This may be called repeatedly to support multiple PSKs. If working with
+    /// large numbers of PSKs, consider using a [`PskSelectionCallback`].
+    ///
+    /// Corresponds to [`s2n_connection_append_psk`].
+    pub fn append_psk(&mut self, psk: &ExternalPsk) -> Result<(), Error> {
+        unsafe {
+            // SAFETY: *mut cast - s2n-tls does not treat the pointer as mutable.
+            s2n_connection_append_psk(self.as_ptr(), psk.as_s2n_ptr() as *mut _).into_result()?
+        };
+        Ok(())
+    }
+
+    /// Corresponds to [`s2n_connection_get_negotiated_psk_identity_length`].
+    pub fn negotiated_psk_identity_length(&self) -> Result<usize, Error> {
+        let mut length = 0;
+        unsafe {
+            s2n_connection_get_negotiated_psk_identity_length(self.connection.as_ptr(), &mut length)
+                .into_result()?
+        };
+        Ok(length as usize)
+    }
+
+    /// Retrieve the negotiated psk identity. Use [`Connection::negotiated_psk_identity_length`]
+    /// to retrieve the length of the psk identity.
+    ///
+    /// Corresponds to [`s2n_connection_get_negotiated_psk_identity`].
+    pub fn negotiated_psk_identity(&self, destination: &mut [u8]) -> Result<(), Error> {
+        unsafe {
+            s2n_connection_get_negotiated_psk_identity(
+                self.connection.as_ptr(),
+                destination.as_mut_ptr(),
+                destination.len() as u16,
```

Comment:
This should *probably* be saturating to u16, not truncating (taking the bottom 16 bits) -- otherwise if I pass you e.g. `1 << 20` as the length of my buffer I'll end up with zero bytes in my buffer.